java/code/src/com/redhat/rhn/domain/server/MinionServer.java

```diff
@@ -219,6 +222,20 @@ public boolean equals(Object other) {
                 .isEquals();
     }
 
+
+    /**
+     * @return Return true when all assigned software channels have valid access tokens.
+     */
+    public boolean hasValidTokensForAllChannels() {
+
+        Set<Channel> tokenChannels = AccessTokenFactory.listByMinion(this)
+                .stream()
```

Comment:
does this require to check it the token is valid or is this done elsewhere already?

Reply:
This is currently used only for a unit test, but it does not check for valid token. Just that every channel has one.
I think it makes sense to check also if the tokens are valid.
I will look into this.